### PR TITLE
Do not throw error when executing empty StableHLO modules

### PR DESCRIPTION
**Pull request by penagos**
This PR addresses a bug introduced by PR #1797 where as a result of the new interpreter API, running empty StableHLO modules would result in an error `Requested main function not found.` whereas previously, the `stablehlo-translate` utility would not throw any error.

(Found by @ghpvnist in #1841).

### CODE REVIEW

**Comment by GleasonK** [comment]
I don't have a strong opinion here, delegating to @ghpvnist. Not sure why someone would want to execute an empty model,  but I suppose that given the StableHLO grammar:

```
Program ::= {Func}
```

The empty module is a valid StableHLO program, so maybe this is a change for the better.

File: stablehlo/reference/Api.cpp [hunk 1]
```diff
@@ -106,6 +106,9 @@ class DefaultInterpreterFallback : public InterpreterFallback {
 llvm::ErrorOr<SmallVector<InterpreterValue>> evalModule(
     ModuleOp module, ArrayRef<InterpreterValue> inputs,
     const InterpreterConfiguration &config) {
+  if (module.getOps<func::FuncOp>().empty())
+    return (SmallVector<InterpreterValue>){};
```
Comment (ghpvnist) [inline stablehlo/reference/Api.cpp:110]: I think this should fix the build failure: `return {};`

**Review by ghpvnist — CHANGES_REQUESTED** [review]

File: stablehlo/reference/Api.cpp (hunk 1 above)
Comment (penagos) [reply stablehlo/reference/Api.cpp]: There's some return type ambiguity in doing this, as it's not possible to know if that's an `llvm::Error` or a `llvm::SmallVector`, so it needs some type hint. Updated.

**Review by ghpvnist — APPROVED** [review]
Thank you for the fix!

**Pull request merged**